### ISSUE TITLE
[a11y] The highlight shape of the favorite website icon does not match the other buttons on the same screen

### ISSUE THREAD

**Issue by hffvld** [brave/brave-browser#53010]
### Description

The highlight shape of the favorite website icon does not match the other buttons on the same screen.

### Steps to reproduce

1. Launch Brave
2. Open few random websites, so `favicons` will appear on the `NTP`
3. Go the tab where any website is opened
4. Tap the URL search bar
5. Tap `favicon` of any website > *Observe*

### Actual result

The highlight shape of the favorite website icon does not match the other buttons on the same screen.

---------------------

| 1 | 2 | 3 |
|:-:|:-:|:-:|
| ![1](https://github.com/user-attachments/assets/ece4c92f-bae9-446e-bc16-fb3270e7b50d) | ![2](https://github.com/user-attachments/assets/430448ca-894f-40dd-83cb-d950b5fdbda6) | ![3](https://github.com/user-attachments/assets/c5b5baa6-4540-4be2-a326-756c9509b617) |

### Expected result

The highlight shape of the favorite website icon must match the other buttons on the same screen.

### Reproduces how often

Easily reproduced

### Brave version

Brave build: 1.89.56 
Chromium: 145.0.7632.109 (Official Build) canary (64-bit) 

### Device

- Pixel 6 / oriole-user 16 BP4A.251205.006 release-keys

### Channel information

- [x] release (stable)
- [x] beta
- [x] nightly

### Reproducibility

- [ ] with Brave Shields disabled
- [ ] with Brave Rewards disabled
- [ ] in the latest version of Chrome

**Comment by aguscruiz** [comment]
Proposing we solve this the following way:


- For icon buttons, let's use the circle shape (I imagine Chromium also uses them so it'd be easier for us) and also our buttons are circle now so that's in line with our standard.

- For the NTP card, we should be rounding those corners. Let's use the same border radius that we're using for the URL bar above and the copy/edit suggestions. Also, we should have some more padding there related to the text. Here's a design spec for this https://www.figma.com/design/q8UcTSAVmrP8jPjoRScJKk/%F0%9F%A4%96-Android-General-Design?node-id=1936-11981&t=roDSb6Lx2Tl111Mo-1
